### PR TITLE
ESLint / Minor bug fixes and eslint-recommended practices

### PR DESCRIPTION
**Pull request by brettz9**
- License: Fix warning about unrecognized SPX identifier in `package.json`: "BSD" -> "BSD-3-Clause"
- Linting: Add recommended ESLInt rules (to catch bugs); 1. Fix: reference to `Point` instead of `Point2D` in `toString`; 2. Avoid globals (`roots`, `cmd`); 3. unnecessary semi-colons; 4. unused variables; 5. Avoid redeclarations; 6. Markup assignment within conditional with extra parentheses
- npm: Add `eslint` script
- npm: Add `package-lock.json`

I'm also planning changing source to ES6 Modules with UMD and ESM distributions (possibly with Babel against other ES6 changes) if you're open to it.

And if you wouldn't mind, I'd also be interested in hearing if you've been tracking the subsequent changes to the original project from which you forked and what led you to fork rather than submit PRs.

And if you have any thoughts about a utility to convert an actual SVG element to your SVG-like constructs.

Thanks!

### CODE REVIEW

**Comment by jfschwarz** [comment]
Great to see you contributing, thank you! :)

To be honest, I cannot quite remember what triggered me to fork the repo instead of just wrapping the kld-intersections modules. What we added on top of the actual intersection calculation functions is an  API to initialize the input parameters for these functions that is straightforward when dealing with SVG elements (see `IntersectionParams.newShape` function). For some reason, forking seemed like the way to go back then, probably because we needed to access private APIs of kld-intersections to realize this.

We didn't do a great job tracking changes from the upstream. The kld-intersections module seems to have come a long way since our fork and we might evaluate whether we could use it as a dependency while keeping our public API. So this module would be nothing more than an API wrapper. We should explore this. Any help would be greatly appreciated ;)

File: .gitignore [hunk 1]
```diff
@@ -6,6 +6,7 @@ lib-cov
 *.out
 *.pid
 *.gz
+package-lock.json
```
Comment (jfschwarz) [inline .gitignore]: The package-log.json is actually meant to be committed to git afaik.

File: test/intersection_tests.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-var bezier = require('../lib/functions/bezier');
+/* var bezier = */ require('../lib/functions/bezier');
```
Comment (jfschwarz) [inline test/intersection_tests.js]: could also remove the entire line

File: .gitignore (hunk 1 above)
Comment (brettz9) [reply .gitignore]: I actually usually do, and you're right, but I've encountered some people lately who don't want it committed. :)  I guess it is between whether one wants predictability across instances (as committing it enforces) and an easier means for consumers getting the latest dependencies without waiting on a `package-lock.json` update. But I will undo this and commit the file.

File: test/intersection_tests.js (hunk 2 above)
Comment (brettz9) [reply test/intersection_tests.js]: I caught that in my other PR, but I've removed first in this one.

**Comment by brettz9** [comment]
I've gone ahead and revised the commits (also in the other PR).

Thank you for publishing this useful library, and thank you for the understanding regarding forking and the new API. I'd also like to build upon your `newShape` code to allow SVG elements themselves to be passed in.

Regarding a new API, what if I were to possibly lint/modularize against their upstream version, and in the process ask about openness to either directly integrating your changes or providing for hooks? I'm not sure how easily/quickly I can gain an insight into your needs to do so, but maybe starting a discussion could point out a direction?

**Pull request merged**